Make confirm alerts show the triggering action

rhian-cs opened the pull request:
## References

> Related Issues/Pull Requests/errors/etc...

Resolves #4015

## Objectives

> What are the objectives of these changes?

When the user clicks on certain buttons in the app, a confirmation alert is shown. However, this alert only asks the user "Are you sure?", but that doesn't give much information about what he should be sure of.

In this PR, I made the confirmation alerts show the triggering action, with the format:

`Are you sure? %{action}`

For instance, if the user clicks on a 'Hide author' button, the alert now displays 'Hide author: Are you sure?'.

System specs were also updated to check if the confirmation alerts actually showed the triggering action.

## Visual Changes

> Any visual changes? please attach screenshots (or gifs) showing them.
> If modified views are public (not the admin panel), try them in mobile display (with your browser's developer console) and add screenshots.

For example, in a page that has comments:

![01-confirmation-screenshot](https://user-images.githubusercontent.com/72531802/121581186-f2135d00-ca03-11eb-9c18-da1aa119ea4e.png)

I didn't include a mobile screenshot because the dev tools still show the alerts as if it was in a computer, but since this is the standard alert, not much has changed.

## Notes

> Mention rake tasks or actions to be done when deploying this changes to a server (if any).
> Explain any caveats, or important things to notice like deprecations (if any).

Since this PR is already pretty big, I plan to open another one (if/after this one is merged) to update the other locales too.

javierm commented:
Hi, @rhian-cs :smile:.

Thank you so much for this pull request, and sorry for the late reply! :pray:.

As part of a list of accessibility improvements regarding the links and buttons in the admin section, we implemented the detailed confirmation dialog in the admin section so it includes the name of the thing it acts on. The general message format is: `"Are you sure? %{action} \"%{name}\""`, so for example it would read like `Are you sure? Hide "Send an expedition to the Sun"` for a proposal titled "Send an expedition to the Sun". See commit 6510cb9615db4.

Would it be possible  to update this pull request to use that format in the public sections? If you can, great! :tada:. If you can't, it's OK; we understand we've kept you waiting for a review far too long.

Thanks!

rhian-cs commented:
Hello @javierm! I'll take another look, this PR was pretty big too.

I hope it doesn't take too long since this branch is pretty outdated as well.

We could perhaps try to redo it or split it into smaller PRs, WDYT?

javierm commented:
@rhian-cs Thanks! :smile: One pull request is OK, if that's fine with you.

Regarding whether it's better to redo it or not, whatever suits you better works for us :wink:.

javierm commented:
(I clicked on "close with comment" instead of "comment" by accident; please ignore :relieved:)

rhian-cs commented:
Hello @javierm!

I've rebased the branch and fixed the failing specs. I also ran all the specs locally. Could you take a look at this and let the CI run the tests?

javierm commented:
Hi, @rhian-cs :smile:.

This is great, thanks! We'll have a look at this pull request before the end of the year :+1:.

Oh, and thank you so much for taking the time to update the tests :wink:.

javierm commented on spec/system/admin/valuator_groups_spec.rb:
```diff
@@ -67,7 +67,10 @@
-    accept_confirm { click_button "Delete" }
+
+    confirmation = "Are you sure? This action will delete "\
+      "\"#{ValuatorGroup.last.name}\" and can't be undone."
```
We run into so-called "flaky tests" (meaning sometimes tests fail with no apparent reason) when running database queries after the first `visit` in the test (see pull request #4475). Here `ValuatorGroup.last` would generate a database query which has a low probability (maybe about 1%?) to generate a `PG::ProtocolViolation` error when running the tests.

How about changing line 67 so we give the group a name like "Early birds"? Then we could write:

```suggestion
      "\Early birds\" and can't be undone."
```

javierm reviewed (commented):
@rhian-cs Thank you so much, this is great! :smile: :tada:

I've left a few comments; let me know what you think! And, by all means,  if you don't agree with something I mention, say so! :wink:

javierm reviewed (approved):
Thanks! :+1: Really appreciate it :smile:.
